src/nemos/glm.py

```diff
@@ -100,28 +124,6 @@ def __init__(
         self._solver_update = None
         self._solver_run = None
```

Comment:
I have a way to add the warning that Alex was talking about. 

- Make the Regularizer stateless (pass the `regularizer_strength` as an input to the two methods)
- Make `regularizer_strength` a property of `BaseRegressor` and move the conversion to float there. Default it to None. Whenever `Ridge`/`Lasso`/`GroupLasso` is set as a regularizer in `BaseRegressor` and `regularizer_strength` is None, raise a warning, set the `regularizer_sregnth` to 1. Whenever regularizer is set to `Unregularized` reset the regularizer strength to None, so that the waning will be triggered again if needed.

What do you think?

Reply:
I think that sounds good